rs/xnet/payload_builder/src/lib.rs

```diff
@@ -530,23 +530,16 @@ impl XNetPayloadBuilderImpl {
         }
 
         if !reject_signals.is_empty() {
-            // TODO(MR-635): Change this check to use the same mechanism used for capping
-            // the number of signals in streams instead.
-            // Given the minimum message size (zero-length sender and receiver, no cycles,
-            // no payload) of 17 bytes; plus 16 bytes for `LabelTree` encoding plus label;
-            // and 16+6 bytes for a `Witness::Known` and a `Witness::Fork` node; we have
-            // a minimum of 55 bytes per encoded message.
-            //
-            // With a `TARGET_STREAM_SIZE_BYTES` of 10 MiB, that means a maximum of just
-            // over 190K messages in a stream. 200K to be conservative.
-            const MAX_STREAM_MESSAGES: u64 = 200_000;
+            // A stream can never have more than `MAX_STREAM_MESSAGES` signals by design,
+            // since we stop pulling messages after reaching this limit. Any subnet with
+            // more than this number of signals can therefore be classified as dishonest.
 
             // An honest subnet will only produce signals for the messages in the incoming
             // stream (i.e. no signals for future messages; and all signals for past
             // messages have been GC-ed). Meaning we can never have signals going back
             // farther than the maximum number of messages in a stream.
             let signals_begin = reject_signals.front().unwrap();
-            if signals_end.get() - signals_begin.index.get() > MAX_STREAM_MESSAGES {
+            if signals_end.get() - signals_begin.index.get() > MAX_STREAM_MESSAGES as u64 {
```

Comment:
Should we go for 2x here to include some wiggle room in case we want to bump this constant in the future (requiring 2x would allow us to bump it by 2x without a staged rollout)?
```suggestion
            if signals_end.get() - signals_begin.index.get() > 2 * MAX_STREAM_MESSAGES as u64 {
```

Reply:
We discussed that at one point and yes we can do that. It's still good enough as a guard against dishonest subnets, but then I also didn't quite see the need for it. I mean all you'd have to do is increase this constant first, which doesn't seem like much of a hurdle.

No strong opinion on my side.

Reply:
On second thought, since the factor of 2 always scales with whatever we set the constant to, this makes it such that we can always change this constant without ever touching this, as long as we don't increase by more than a factor of 2, which seems unlikely. So why not let's go that way then.